src/core/net/ip6_address.hpp

```diff
@@ -324,6 +324,17 @@ class Address : public otIp6Address, public Equatable<Address>, public Clearable
         kGlobalScope         = 14, ///< Global scope
     };
 
+    /**
+     * This enumeration defines IPv6 address type filter.
+     *
+     */
+    enum TypeFilter : uint8_t
+    {
+        kTypeAny,           ///< Accept any IPv6 address (unicast or multicast).
```

Comment:
```suggestion
        kTypeFilterAny,           ///< Accept any IPv6 address (unicast or multicast).
```
Put type name in the value names?

Reply:
I avoided this mainly since I think/feel `kTypeFilterUnicastOnly` reads as if we are filtering out the unicast addresses.

Reply:
I am fine with kTypeXxx